pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,506 @@
+
+    num_wires = qml.operation.AnyWires
+    grad_method = "A"  # supports analytic gradients
+    batch_size = None
+    ndim_params = None  # could be (0,) * len(coeffs), but it is not needed. Define at class-level
+
+    def _flatten(self):
+        # note that we are unable to restore grouping type or method without creating new properties
+        return (self._coeffs, self._ops, self.data), (self.grouping_indices,)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        new_op = cls(data[0], data[1])
+        new_op._grouping_indices = metadata[0]  # pylint: disable=protected-access
+        new_op.data = data[2]
+        return new_op
+
+    def __init__(
+        self,
+        coeffs,
+        observables: List[Operator],
+        simplify=False,
+        grouping_type=None,
+        method="rlf",
+        _pauli_rep=None,
+        id=None,
+    ):
+        if qml.math.shape(coeffs)[0] != len(observables):
+            raise ValueError(
+                "Could not create valid LinearCombination; "
+                "number of coefficients and operators does not match."
+            )
+
+        self._coeffs = coeffs
+
+        self._ops = [convert_to_opmath(op) for op in observables]
+
+        self._hyperparameters = {"ops": self._ops}
+
+        self._grouping_indices = None
+
+        with qml.QueuingManager().stop_recording():
+            operands = [qml.s_prod(c, op) for c, op in zip(coeffs, observables)]
+
+        # TODO use grouping functionality of Sum once https://github.com/PennyLaneAI/pennylane/pull/5179 is merged
+        super().__init__(
+            *operands, grouping_type=grouping_type, method=method, id=id, _pauli_rep=_pauli_rep
+        )
+
+        if simplify:
+            # TODO clean up this logic, seems unnecesssarily complicated
+
+            simplified_coeffs, simplified_ops, pr = self._simplify_coeffs_ops()
+
+            self._coeffs = (
+                simplified_coeffs  # Losing gradient in case of torch interface at this point
+            )
+
+            self._ops = simplified_ops
+            with qml.QueuingManager().stop_recording():
+                operands = [qml.s_prod(c, op) for c, op in zip(self._coeffs, self._ops)]
+
+            super().__init__(*operands, id=id, _pauli_rep=pr)
+
+    def _check_batching(self):
+        """Override for LinearCombination, batching is not yet supported."""
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        decimals = None if (len(self.parameters) > 3) else decimals
+        return super(CompositeOp, self).label(
+            decimals=decimals, base_label=base_label or "𝓗", cache=cache
+        )  # Skipping the label method of CompositeOp # pylint: disable=bad-super-call
+
+    @property
+    def coeffs(self):
+        """Return the coefficients defining the LinearCombination.
+
+        Returns:
+            Iterable[float]): coefficients in the LinearCombination expression
+        """
+        return self._coeffs
```

Comment:
Are we having deprecation cycles for these?

Reply:
I wasnt quite sure about this yet as `data` does look different for `CompositeOp`s and isnt just the trainable parameters. In particular for a mix with Hermitians you get something like this:
```python
coeffs = [1., 1., 1., 1.]
ops = [
    X(0),
    qml.Hermitian(np.eye(2), wires=[0]),
    X(1),
    qml.Hermitian(np.eye(2), wires=[0]),
]
ops = qml.ops.LinearCombination(coeffs, ops)
ops.data
```

```
(1.0,
 1.0,
 array([[1., 0.],
        [0., 1.]]),
 1.0,
 1.0,
 array([[1., 0.],
        [0., 1.]]))
```

same for parameters 😬

Reply:
`op.terms()`?

Reply:
but isnt that just returning `op._coeffs` again? Or do you mean keep the private attribute but just not make it user facing?